### PR TITLE
bash completion: fixed invalid identifier

### PR DESCRIPTION
**Pull request by alalazo**

### CODE REVIEW

File: share/spack/spack-completion.bash [hunk 1]
```diff
@@ -288,7 +288,7 @@ function _spack_debug {
     fi
 }
 
-function _spack_create-db-tarball {
+function _spack_create_db_tarball {
```
Comment (adamjstewart) [inline share/spack/spack-completion.bash]: Wouldn't it actually be `_spack_debug_create-db-tarball`?

Comment (alalazo) [inline share/spack/spack-completion.bash]: @adamjstewart On rhel6 I get:
```
$ . share/spack/setup-env.sh 
-sh: `_spack_create-db-tarball': not a valid identifier
```
Honestly I didn't check how completion works. I thought it was a typo.

Comment (adamjstewart) [inline share/spack/spack-completion.bash]: Hmm... I don't get that on my CentOS 6 box. What shell are you using? What version?

Comment (alalazo) [inline share/spack/spack-completion.bash]:
```console
-sh-4.1$ which $SHELL
/bin/sh
-sh-4.1$ ls -l /bin/sh
lrwxrwxrwx 1 root root 4 18 lug  2016 /bin/sh -> bash
-sh-4.1$ bash --version
GNU bash, version 4.1.2(1)-release (x86_64-redhat-linux-gnu)
Copyright (C) 2009 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
-sh-4.1$ which bash
```

Comment (adamjstewart) [inline share/spack/spack-completion.bash]: Ah, I know what the problem is. When invoked as `sh`, `bash` will run in compatibility mode. In POSIX, the only valid characters in function names are underscores and alphanumeric characters. So I think what you'll need to do is change it to `_spack_debug_create_db_tarball` and modify the `subfunction` variable to replace dashes with underscores.

**Pull request merged**